Code/ForceField/MMFF/Params.cpp

```diff
@@ -55,25 +69,34 @@ MMFFAromCollection::MMFFAromCollection(const boost::uint8_t *mmffArom) {
 const boost::uint8_t defaultMMFFArom[] = {37, 38, 39, 44, 58, 59, 63, 64, 65,
                                           66, 69, 76, 78, 79, 80, 81, 82};
 
-class MMFFDefCollection *MMFFDefCollection::ds_instance = NULL;
+boost::scoped_ptr<MMFFDefCollection> MMFFDefCollection::ds_instance;
+#ifdef BOOST_THREAD_PROVIDES_ONCE_CXX11
+boost::once_flag MMFFDefCollection::ds_flag;
+#else
+boost::once_flag MMFFDefCollection::ds_flag = BOOST_ONCE_INIT;
+#endif
 
 extern const std::string defaultMMFFDef;
 
 MMFFDefCollection *MMFFDefCollection::getMMFFDef(const std::string &mmffDef) {
-  if (!ds_instance) {
-    ds_instance = new MMFFDefCollection(mmffDef);
-  } else if (mmffDef != "") {
-    delete ds_instance;
-    ds_instance = new MMFFDefCollection(mmffDef);
+#ifdef RDK_THREADSAFE_SSS
+  boost::call_once(ds_flag, boost::bind(&create, mmffDef));
```

Comment:
Brian is right. The only proper way to handle this is through a flyweight, as we cannot create a whole new set of custom parameters for each thread asking for it if that set already exists, as that would quickly become a huge leak if called repeatedly. I'll revisit this PR as time allows.